addons/missileguidance/functions/fnc_guidancePFH.sqf

```diff
@@ -18,95 +18,125 @@
     };
-    if (_adjustVectorY < 0) then {
-        _roll = - ( (_minDeflection max ((abs _adjustVectorY) min _maxDeflection) ) );
-    } else {
-        if (_adjustVectorY > 0) then {
-            _roll = ( (_minDeflection max (_adjustVectorY min _maxDeflection) ) );
-        };
+
+    if (GVAR(debug_drawGuidanceInfo)) then {
+        private _projectilePosAGL = ASLToAGL _projectilePos;
+        drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\selectover_ca.paa", [1,0,0,1], _projectilePosAGL vectorAdd [0, 0, 1], 0.75, 0.75, 0, str _commandedAcceleration, 1, 0.025, "TahomaB"];
+        drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\selectover_ca.paa", [1,1,0,1], _projectilePosAGL vectorAdd [0, 0, 2], 0.75, 0.75, 0, _navigationType, 1, 0.025, "TahomaB"];
+        drawLine3D [_projectilePosAGL, _projectilePosAGL vectorAdd _commandedAcceleration, [1, 0, 1, 1]];
     };
-    if (_adjustVectorZ < 0) then {
-        _pitch = - ( (_minDeflection max ((abs _adjustVectorZ) min _maxDeflection) ) );
-    } else {
-        if (_adjustVectorZ > 0) then {
-            _pitch = ( (_minDeflection max (_adjustVectorZ min _maxDeflection) ) );
+
+    // activate missile servos and change direction
+    if (!isGamePaused && accTime > 0) then {
+        _guidanceParameters params ["_yaw", "_roll", "_pitch"];
+
+        _commandedAcceleration = _projectile vectorWorldToModelVisual _commandedAcceleration;
+        _commandedAcceleration params ["_yawChange", "", "_pitchChange"];
+
+        if (isNil "_yawChange") then {
+            _yawChange = 0;
+        };
+        if (isNil "_pitchChange") then {
+            _pitchChange = 0;
+        };
+
+        private _clampedPitch = (_pitchChange min _pitchRate) max -_pitchRate;
+        private _clampedYaw = (_yawChange min _yawRate) max -_yawRate;
+        
+        // controls are either on or off, no proportional
+        if (_isBangBangGuidance) then {
+            private _pitchSign = if (_clampedPitch == 0) then {
+                0
+            } else {
+                _clampedPitch / abs _clampedPitch
+            };
+            private _yawSign = if (_clampedYaw == 0) then {
+                0
+            } else {
+                _clampedYaw / abs _clampedYaw
+            };
+            _clampedPitch = _pitchSign * 50;
+            _clampedYaw = _yawSign * 50;
         };
-    };
-    private _finalAdjustVector = [_yaw, _roll, _pitch];
 
-    TRACE_3("", _pitch, _yaw, _roll);
-    TRACE_3("", _targetVector, _adjustVector, _finalAdjustVector);
+        TRACE_9("pitch/yaw/roll",_pitch,_yaw,_roll,_yawChange,_pitchChange,_pitchRate,_yawRate,_clampedPitch,_clampedYaw);
 
-    if (accTime > 0) then {
-        private _changeVector = (vectorDir _projectile) vectorAdd _finalAdjustVector;
-        TRACE_2("",_projectile,_changeVector);
-        [_projectile, _changeVector] call FUNC(changeMissileDirection);
+        _pitch = _pitch + _clampedPitch * _timestep;
+        _yaw = _yaw + _clampedYaw * _timestep;
```

Comment:
I think this will have problems with gimble lock
e.g. a bomb dropping almost straight down should be able to easily make large yaw change

another way to do this would be with vector rotation, something like
`[velocity _projectile, (velocity _projectile) vectorCrossProduct _commandDir, _yaw] call CBA_fnc_vectRotate3D;`

Reply:
I solved this by using quaternions. I am like 90% sure this fixes it and doesn't mess anything else up